tools/checkpoint_conversion/convert_bert_checkpoints.py

```diff
@@ -0,0 +1,631 @@
+                                'MODEL_TYPE' : "bert_medium",
+                             'MODEL_SUFFIX' : "uncased",
+                             'MODEL_SPEC_STR' : "L-8_H-512_A-8",
+                             'VOCAB_SIZE' :  30522,
+                             'NUM_LAYERS' : 8,
+                             'NUM_ATTN_HEADS' :  8,
+                             'EMBEDDING_SIZE' :  512
+                               },
+    "bert_small_uncased_en":{'base':"roberta.base",
+                             'base_model':"bert_small_uncased_en",
+                             'MODEL_TYPE' : "bert_small",
+                             'MODEL_SUFFIX' : "uncased",
+                             'MODEL_SPEC_STR' : "L-4_H-512_A-8",
+                             'VOCAB_SIZE' :  30522,
+                             'NUM_LAYERS' : 4,
+                             'NUM_ATTN_HEADS' :  8,
+                             'EMBEDDING_SIZE' :  512},
+    "bert_tiny_uncased_en": {'base':"roberta.base",
+                             'base_model':"bert_tiny_uncased_en",
+                             'MODEL_TYPE' : "bert_tiny",
+                             'MODEL_SUFFIX' : "uncased",
+                             'MODEL_SPEC_STR' : "L-2_H-128_A-2",
+                             'VOCAB_SIZE' :  30522,
+                             'NUM_LAYERS' : 2,
+                             'NUM_ATTN_HEADS' :  2,
+                             'EMBEDDING_SIZE' :  128}
+}
+
+flags.DEFINE_string(
+    "preset", None, f'Must be one of {",".join(PRESET_MAP.keys())}'
+)
+
+
+def download_model(preset):
+    print("-> Download original weights in " + preset)
+    zip_path = f"""https://storage.googleapis.com/tf_model_garden/nlp/bert/v3/{MODEL_SUFFIX}_L-12_H-768_A-12.tar.gz"""
+    path_file = keras.utils.get_file(
+        f"""{preset}.tar.gz""",
+        zip_path,
+        extract=False,
+        archive_format="tar",
+    )
+    print('path_file', path_file)
+    extract_dir = f"./content/{preset}/tmp/temp_dir/raw/"
+    vocab_path = os.path.join(extract_dir, "vocab.txt")
+    checkpoint_path = os.path.join(extract_dir, "bert_model.ckpt")
+    config_path = os.path.join(extract_dir, "bert_config.json")
+
+    os.system(f"tar -C ./content/{preset} -xvf {path_file}")
+
+
+    return vocab_path, checkpoint_path, config_path
+
+
+def convert_checkpoints(preset,checkpoint_path,config_dict):
+    print("\n-> Convert original weights to KerasNLP format.")
+
+    # Transformer layers.
+    vars = tf.train.list_variables(checkpoint_path)
+    weights = {}
+    for name, shape in vars:
+        weight = tf.train.load_variable(checkpoint_path, name)
+        weights[name] = weight
+        print(name)
+    model = keras_nlp.models.BertBackbone.from_preset("bert_tiny_en_uncased",
+                                                      load_weights=True)  # keras_nlp.models.BertBase(vocabulary_size=VOCAB_SIZE)
+    model.summary()
+    if preset in ['bert_base_en_uncased', 'bert_base_en']:
```

Comment:
I don't think we should need all these if cases based on preset. Why do we need to add all these?

It's probably easiest to write this for checkpoints from a single source. The official BERT repo has all checkpoints we need. So you could model on https://github.com/keras-team/keras-nlp/blob/master/tools/checkpoint_conversion/bert_tiny_uncased_en.ipynb, for example.

Reply:
here we don't download from a fixed source f"""https://storage.googleapis.com/bert_models/2020_02_20/{MODEL_SUFFIX}_{MODEL_SPEC_STR}.zip"""

Reply:
@mattdangerw - I will push some changes later today, will use ckpts from a single source

Reply:
@abheesht17 any update on this?